Tribler/Core/TorrentChecker/session.py

```diff
@@ -475,8 +474,12 @@ def remove_transaction_id(self):
         if self in UdpTrackerSession._active_session_dict:
             del UdpTrackerSession._active_session_dict[self]
 
-        if self.transaction_id in self.socket_mgr.tracker_sessions:
-            self.socket_mgr.tracker_sessions.pop(self.transaction_id)
+        # Checking for socket_mgr is a workaround for race condition
+        # in Tribler Session startup/shutdown that sometimes causes
+        # unit tests to fail on teardown.
+        if self.socket_mgr:
```

Comment:
Please put this on the same line: `if self.socket_mgr and self.transaction_id in self.socket_mgr.tracker_sessions`

Reply:
OK